### ISSUE TITLE
Metric in train/test phases

### ISSUE THREAD

**Issue by AntonioCarta** [ContinualAI/avalanche#295]
Metrics can be computed computed during train or test phases. However, these phases are not equal. During test, data is left as is and the model is fixed during the entire loop, while during test:
- the model changes (SGD)
- the data may be augmented (replay)

This means that metrics computed during train phases are approximation of the true metrics (which can be completely wrong for large replay size).

I propose two changes. First, most metrics computed during train need a different name (for the `MetricResult`, not the class), such as `RunningAccuracy` for train and `Accuracy` for test.

Example output now:
```
TRAIN
Epoch 3 ended.
        Loss/Train/Task000 = 0.1704
        Loss_MB/Train/Task000 = 0.1268
        Top1_Acc_Epoch/Train/Task000 = 0.9458
        Top1_Acc_Minibatch/Train/Task000 = 0.9630

TEST
> Test on step 4 (Task 0) ended.
        Loss/Test/Task000 = 0.0663
        Loss_MB/Test/Task000 = 0.1009
```

Example output proposed:
```
TRAIN
Epoch 3 ended.
        RunningLoss/Train/Task000 = 0.1704
        RunningLoss_MB/Train/Task000 = 0.1268
        RunningTop1_Acc_Epoch/Train/Task000 = 0.9458
        RunningTop1_Acc_Minibatch/Train/Task000 = 0.9630

TEST
> Test on step 4 (Task 0) ended.
        Loss/Test/Task000 = 0.0663
        Loss_MB/Test/Task000 = 0.1009
```

There is also an additional problem. The `test` method is really an evaluation method. I can call `strategy.test(train_stream)` if I want to evaluate metrics on the train stream and have correct results (not the running approximation).
I propose two changes to solve this issue:
- change `strategy.test` in `strategy.eval`. We already have `model.eval()` and `data.eval()` so it seems intuitive to me.
- give data streams a name, so that you can do `step_info.name` to find the split name.

The second point is needed because right now you can distinguish between `strategy.test(train_stream)` and `strategy.test(test_stream)`.

With this additional change the output becomes:
```
TRAIN
Epoch 3 ended.
        RunningLoss/Train/Task000 = 0.1704
        RunningLoss_MB/Train/Task000 = 0.1268
        RunningTop1_Acc_Epoch/Train/Task000 = 0.9458
        RunningTop1_Acc_Minibatch/Train/Task000 = 0.9630

EVAL
> Eval on TRAIN stream step 4 (Task 0) ended.
        Loss/Train/Task000 = 0.0663
        Loss_MB/Train/Task000 = 0.1009
> Eval on TEST stream step 4 (Task 0) ended.
        Loss/Test/Task000 = 0.0663
        Loss_MB/Test/Task000 = 0.1009
```
